components/resources/library/src/webMain/kotlin/org/jetbrains/compose/resources/ResourceState.web.kt

```diff
@@ -16,7 +16,7 @@ internal actual fun <T> rememberResourceState(
 ): State<T> {
     val environment = LocalComposeEnvironment.current.rememberEnvironment()
     val scope = rememberCoroutineScope()
-    return remember(key1) {
+    return remember(key1, environment) {
```

Comment:
Added a `rememberResourceStateAffectedByEnvironmentChanges` test for this change